manager_janitor.go

```diff
@@ -335,6 +335,111 @@ func (mgr *Manager) pindexesRestart(
+		err := os.Rename(pi.pindex.Path, newPath)
+		if err != nil {
+			// removed cleanDir() for old path since in case there is a renaming error,
+			// the user would not want to lose existing pindex data for hibernation status
+			// changes.
+			return nil, fmt.Errorf("janitor: restartPIndex"+
+				" updating pindex: %s path: %s failed, err: %v",
+				pi.pindex.Name, newPath, err)
+		}
+		// removed checks for old path still existing/new path not created since if that
+		// does happen, it will be reflected as a renaming error.
+	}
+	pi2 := pi.pindex.Clone() //pi2 is the new pindex.
+	pi2.Name = pi.planPIndexName
+	pi2.Path = newPath
+
+	// update the new indexdef param changes
+	buf, err := json.Marshal(pi2)
+	if err != nil {
+		return nil, fmt.Errorf("janitor: tempRestartPIndex"+
+			" Marshal pindex: %s, err: %v", pi2.Name, err)
+
+	}
+	err = ioutil.WriteFile(pi2.Path+string(os.PathSeparator)+
+		PINDEX_META_FILENAME, buf, 0600)
+	if err != nil {
+		return nil, fmt.Errorf("janitor: tempRestartPIndex could not save "+
+			"PINDEX_META_FILENAME,"+" path: %s, err: %v", pi2.Path, err)
+	}
+	// renaming only after writing new pindex meta
+
+	if complete {
+		if pi2.closed {
+			pi2.closed = false
+		}
+		restart := func() {
+			go restartPIndex(mgr, pi2)
+		}
+		impl, dest, err := OpenPIndexImplUsing(pi2.IndexType, pi2.Path,
+			pi2.IndexParams, restart)
+		if err != nil {
+			return nil, fmt.Errorf("pindex: could not open indexType: %s,"+
+				" path: %s, err: %v", pi2.IndexType, pi2.Path, err)
+		}
+
+		pi2.Impl = impl
+		pi2.Dest = dest
+	}
+
+	err = mgr.registerPIndex(pi2)
+	if err != nil {
+		log.Printf("error in registering pindex %s", pi2.Name)
+		return nil, fmt.Errorf("janitor: tempRestartPIndex failed to "+
+			"register pindex: %s, err: %v", pi2.Name, err)
+	}
+	return pi2, nil
+}
+
+// This function hibernates pindexes i.e. closes them without removing the related files.
+func (mgr *Manager) hibernatePIndex(pi *pindexStatusChangeReq, complete bool) error {
+	log.Printf("janitor: hibernating pindex %s", pi.pindex.Name)
+	return pi.pindex.Close(false)
+}
+
+// This function is a wrapper function for the specific functions used for
+// transitioning pindexes, based on the request type.
+func (mgr *Manager) transitionPIndex(pi *pindexStatusChangeReq) error {
+	err := mgr.stopPIndex(pi.pindex, false)
+	if err != nil {
+		return fmt.Errorf("janitor: error closing pindex %s: %e",
+			pi.pindex.Name, err)
+	}
+	var complete bool
+	if pi.reqType == PHASE_CHANGE {
+		complete = true // complete activation required
+	} else {
+		complete = false
+	}
+	log.Printf("pindex %s being activated", pi.pindex.Name)
+	pi.pindex, err = mgr.activatePIndex(pi, complete)
+	if err != nil {
+		return fmt.Errorf("janitor: error activating pindex: %s",
+			err.Error())
+	}
+	if pi.reqType == HIBERNATE_PINDEX {
+		log.Printf("pindex %s being hibernated", pi.pindex.Name)
+		err = mgr.hibernatePIndex(pi, complete)
```

Comment:
Dont follow why do you need this?
HIBERNATE_PINDEX - was never opened during activatePIndex??